lib/datasource.js

```diff
@@ -194,6 +194,11 @@ util.inherits(DataSource, EventEmitter);
 // allow child classes to supply a data access object
 DataSource.DataAccessObject = DataAccessObject;
 
+/**
+ * Global maximum number of event listeners
+ */
+DataSource.prototype.DEFAULT_MAX_OFFLINE_REQUESTS = 16;
```

Comment:
You can set at class level and reference it as `DataSource.DEFAULT_MAX_OFFLINE_REQUESTS`.

```js
DataSource.DEFAULT_MAX_OFFLINE_REQUESTS = 16;
```

Reply:
I tried this in [this commit](https://github.com/strongloop/loopback-datasource-juggler/pull/1767/commits/b14875da2a24b81821135075d4753d5e1092bef5) but now I can only access it through `this.juggler.DataSource.DEFAULT_MAX_OFFLINE_REQUESTS`/`this.juggler.Schema.DEFAULT_MAX_OFFLINE_REQUESTS` instead of `this.DEFAULT_MAX_OFFLINE_REQUESTS`. I think we should keep it with `prototype`, but what do you think?

Reply:
You should be able to access the setting as `DataSource.DEFAULT_MAX_OFFLINE_REQUESTS`. What's your concern?

Reply:
So we don't use `this.DEFAULT_MAX_OFFLINE_REQUESTS`, we use `DataSource.DEFAULT_MAX_OFFLINE_REQUESTS`? I thought `this` was equivalent to `DataSource` in this file, which is where my confusion was stemming from. I understand now though, thanks!